### ISSUE TITLE
BigDecimal conversion problem - morphia 1.3 - MongoDB 3.4

### ISSUE THREAD

**Issue by creusar** [mongodb/morphia#1093]
Hello, i'm using MongoDB 3.4 and morphia 1.3 to test the new Decimal128 data type. I use to have a custom TypeConverter for my BigDecimals, they were stored on the database as strings.

So far i've disabled my custom converter, and updated the data directly on MongoDB. I can't make morphia save my BigDecimals as Decimal128. Inspecting the execution seems like the Mapper doesn't get the desired converter to BigDecimal, instead it uses the default Embedded Mapper.

The class for my test:

```java
public class Main {

    public static void main(String[] args) {
        Morphia morphia = new Morphia();
        Datastore datastore = morphia.createDatastore(new MongoClient("localhost"), "bigdecimal");
        Foo foo = new Foo();
        foo.setNumber(new BigDecimal("0.92348237942346239"));
        datastore.save(foo);
    }
}

@Entity
class Foo {

    @Id
    private ObjectId id;
    private BigDecimal number;

    public ObjectId getId() {
        return id;
    }

    public void setId(ObjectId id) {
        this.id = id;
    }

    public BigDecimal getNumber() {
        return number;
    }

    public void setNumber(BigDecimal number) {
        this.number = number;
    }

}
```

And a database `db.Foo.find()`:

```
{
        "_id" : ObjectId("5846fa9aa618770ee4f115d7"),
        "className" : "Foo",
        "number" : {
                "intVal" : {
                        "signum" : 1,
                        "mag" : [
                                21501499,
                                -1372597761
                        ],
                        "bitCount" : 0,
                        "bitLength" : 0,
                        "lowestSetBit" : 0,
                        "firstNonzeroIntNum" : 0
                },
                "scale" : 17
        }
}
```

Am I missing some configuration for the conversion BigDecimal <-> Decimal128 work properly?

**Comment by evanchooly** [comment]
Oh, dear.  That's definitely a bug.  A missing super constructor call.  Turns out we tested all the conversion scenarios by invoking the converter directly and never tried round tripping that conversion in an entity.  mea culpa.

**Comment by creusar** [comment]
Nice! Missed that too, my custom converter had the superconstructor call.

Thanks for the quick fix, any estimate when the v1.3.1 will be available on maven?

**Comment by evanchooly** [comment]
Probably next week at this point.  There's a handful of minor updates I'd like to take a look at first.
